src/Evolution/Systems/Cce/BoundaryData.hpp

```diff
@@ -221,4 +333,218 @@ void dyads(
+ * \partial_\lambda U = - \left(g^{\lambda A}_{,\lambda}
+ * + \frac{r_{,\lambda A}}{r_{,\lambda}} g^{A B}
+ * + \frac{r_{, B}}{r_{, \lambda}} g^{A B}_{, \lambda}\right)q_A
+ * + 2 \beta_{, \lambda}(U + g^{\lambda A} q_A)
+ * \f]
+ *
+ * and
+ *
+ * \f[
+ * Q = r^2 (J \partial_\lambda \bar U + K \partial_\lambda U)
+ * \f]
+ *
+ * also provided is \f$\partial_r U\f$, which is separately useful to cache for
+ * other intermediate steps in the CCE computation.
+ */
+void bondi_q_worldtube_data(
+    gsl::not_null<Scalar<SpinWeighted<ComplexDataVector, 1>>*> bondi_q,
+    gsl::not_null<Scalar<SpinWeighted<ComplexDataVector, 1>>*> dr_bondi_u,
+    const Scalar<DataVector>& d2lambda_r,
+    const tnsr::AA<DataVector, 3, Frame::RadialNull>&
+        dlambda_inverse_null_metric,
+    const tnsr::a<DataVector, 3, Frame::RadialNull>& d_bondi_r,
+    const tnsr::i<ComplexDataVector, 2, Frame::RadialNull>& down_dyad,
+    const tnsr::i<DataVector, 3, Frame::RadialNull>& angular_d_dlambda_r,
+    const tnsr::AA<DataVector, 3, Frame::RadialNull>& inverse_null_metric,
+    const Scalar<SpinWeighted<ComplexDataVector, 2>>& bondi_j,
+    const Scalar<SpinWeighted<ComplexDataVector, 0>>& bondi_r,
+    const Scalar<SpinWeighted<ComplexDataVector, 1>>& bondi_u) noexcept;
+
+/*!
+ * \brief Compute the Bondi metric contribution \f$(\partial_u J)_{y} \equiv
+ * H\f$ on the worldtube boundary.
+ *
+ * \details The numerical time derivative (along the worldtube, rather than
+ * along the surface of constant Bondi \f$r\f$) is computed by (see equation
+ * (48) of \cite Barkett2019uae)
+ *
+ * \f[
+ * (\partial_u J)_y = \frac{1}{2 r^2} q^A q^B \partial_u g_{A B}
+ * - \frac{2 \partial_u r}{r} J
+ * \f]
+ *
+ * \note There is the regrettable notation difference with the primary reference
+ * for these formulas \cite Barkett2019uae in that we denote with \f$H\f$ the
+ * time derivative at constant numerical radius, where \cite Barkett2019uae uses
+ * \f$H\f$ to denote the time derivative at constant Bondi radius.
+ */
+void bondi_h_worldtube_data(
+    gsl::not_null<Scalar<SpinWeighted<ComplexDataVector, 2>>*> bondi_h,
+    const tnsr::a<DataVector, 3, Frame::RadialNull>& d_bondi_r,
+    const Scalar<SpinWeighted<ComplexDataVector, 2>>& bondi_j,
+    const tnsr::aa<DataVector, 3, Frame::RadialNull>& du_null_metric,
+    const Scalar<SpinWeighted<ComplexDataVector, 0>>& bondi_r,
+    const tnsr::I<ComplexDataVector, 2, Frame::RadialNull>& up_dyad) noexcept;
+
+/*!
+ * \brief Compute the Bondi metric contribution \f$(\partial_u J)_r\f$ on the
+ * worldtube boundary.
+ *
+ * \details The numerical time derivative (along the surface of constant r, not
+ * along the worldtube) is computed by (see equation (50) of
+ * \cite Barkett2019uae)
+ *
+ * \f[
+ * \partial_u J = \frac{1}{2 r^2} q^A q^B \left(\partial_u g_{A B}
+ * - \frac{r_{,u}}{r_{, \lambda}} \partial_\lambda g_{A B}\right)
+ * \f]
+ *
+ * \note There is the regrettable notation difference with the primary reference
```

Comment:
there is no `H` in this formula?

Reply:
Yes, but this is the one the reference calls `H`, so I felt the clarifying note should be present.